fastlane/lanes/build.rb

```diff
@@ -80,31 +82,61 @@
     )
   end
 
-  # Builds the app and uploads it to TestFlight / App Store Connect
+  # Builds the WordPress app and uploads it to TestFlight, for beta-testing or final release
   #
   # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
-  # @option [Boolean] create_gh_release If true, creates a GitHub Release draft after the upload, with zipped xcarchive as artefact
+  # @option [Boolean] skip_prechecks (default: false) If true, don't run the ios_build_prechecks and ios_build_preflight
+  # @option [Boolean] create_release If true, creates a GitHub Release draft after the upload, with zipped xcarchive as artefact
   # @option [Boolean] beta_release If true, the GitHub release will be marked as being a pre-release
   #
   # @called_by CI
-  # @calls build_and_upload_itc
   #
   desc 'Builds and uploads for distribution to App Store Connect'
   lane :build_and_upload_app_store_connect do |options|
-    ios_build_prechecks(
-      skip_confirm: options[:skip_confirm],
-      internal: options[:beta_release],
-      external: true
+    ios_build_prechecks(skip_confirm: options[:skip_confirm], external: true) unless options[:skip_prechecks]
+    ios_build_preflight unless options[:skip_prechecks]
+
+    sentry_check_cli_installed
+    appstore_code_signing
+
+    gym(
+      scheme: 'WordPress',
+      workspace: WORKSPACE_PATH,
+      clean: true,
+      export_team_id: get_required_env('EXT_EXPORT_TEAM_ID'),
+      output_directory: BUILD_PRODUCTS_PATH,
+      derived_data_path: DERIVED_DATA_PATH,
+      export_options: { method: 'app-store' }
```

Comment:
Nitpick on argument's position, since I saw you addressed [the same suggestion](https://github.com/wordpress-mobile/WordPress-iOS/pull/18140#discussion_r832798229) in #18140:

```suggestion
      output_directory: BUILD_PRODUCTS_PATH,
      derived_data_path: DERIVED_DATA_PATH,
      export_team_id: get_required_env('EXT_EXPORT_TEAM_ID'),
      export_options: { method: 'app-store' }
```

Also, given this works without the explicit `export_method` parameter, maybe we can do without it in the enterprise version, too? 🤔 Or maybe this works merely because `'app-store'` is the value `gym` falls back to internally (I checked `action gym` and the docs don't specify a default).

Just a thought, anyway, maybe something to verify in a standalone PR later. Fine tuning these when we don't need to has diminishing returns...

Reply:
I vote we keep it explicit. Here's why:

 - I think [this portion of code in fastlane](https://github.com/fastlane/fastlane/blob/668d976ec61d74509f3123ea1b7fe1836a1052e4/fastlane/lib/fastlane/actions/build_app.rb#L16) might be what sets the `export_method` if not set; i.e. to my understanding, if unset, it derives it from `Actions.lane_context[SharedValues::SIGH_PROFILE_TYPE]` (unless that value is `development`); and that value is set by `match`; so the fact that we run match before running that command is probably why it ends up picking the right value.
 - At some point I thought that `fastlane` would use the `export_method` parameter to inject it in `export_options: { method: … }` if it was not defined there already; that would make sense to me… though not sure how they would decide to do this in one direction vs the other (should `export_method` be the source of truth, or should `export_options: { method: }` be?). [This portion of the code](https://github.com/fastlane/fastlane/blob/668d976ec61d74509f3123ea1b7fe1836a1052e4/fastlane/lib/fastlane/actions/build_app.rb#L33) looked like it could be what they were doing here… but then reading from [explanations in this method's implementation](https://github.com/fastlane/fastlane/blob/668d976ec61d74509f3123ea1b7fe1836a1052e4/gym/lib/gym/code_signing_mapping.rb#L16) I don't think it does (it only uses it to check if it's present and help resolve the profiles to use, nothing more?)…

---

I'd vote to keep the `export_method` and continue to set it explicitly in both places, rather than relying on what other actions (`match`) was potentially called before (especially because we could imagine one day calling `match` for all types (appstore+enterprise) all in some setup step common to multiple lanes, breaking that guaranteed that the last call to `match` before this call to `gym` happened to be about the `export-method` we want to use.

We _could_ consider DRYing the value from `export_method: …` with the one from `export_options: { method: … }`… but given that it would be a very localized DRY-ing for each separate call of `gym`, I'm not convinced it's worth it compared to just repeat the `'app-store` literal 🤷